### DISCUSSION TITLE
Subsetting aitchison distance

### DISCUSSION THREAD

**Discussion by jorondo1** [david-barnett/microViz#107] [Q&A]
Hi,
Following the recommendations in the doc for PERMANOVA using aitchison distances (option 2): 

`pass an object with untransformed (or 'identity' transformed) taxa to the data argument of dist_calc() and specify dist = "aitchison".
`

is it ok to subset the resulting psExtra object, i-e remove certain low abundance taxa, before running dist_permanova ? I thought it would, but I realise that this procedure doesnt rcclr transform the data. I thought aitchison distance calculations were based on log-ratio transformed compositions, so I'm a bit confused. I was counting on subcompositional coherence to subset, but my (very basic) understanding of the underlying maths of compositional data may be causing my confusion.

Looking forward to your input

**Comment by david-barnett** [comment]
Hi Jonathan 

You ask 
> is it ok to subset the resulting psExtra object, i-e remove certain low abundance taxa, before running dist_permanova ?

I am not sure what you mean by "resulting psExtra"? 

I'll assume you mean the output of dist_calc(dist = "aitchison"). Correct me if I'm wrong?

Now some possibly helpful pieces of info:

- Aitchison distances are equivalent to euclidean distances on CLR transformed taxa (assuming you use the same pseudocount or zero-replacement strategy each time)
- If you want clr-transformed taxon abundances in a psExtra, run tax_transform(trans = "clr")
- You could, from a practical standpoint, filter taxa before or after CLR transformation (if after, accessing the original counts that psExtra objects retain after transformation, with use_counts = TRUE in tax_filter)
- I would not filter taxa **after distance calculation**, as then the distances aren't relevant to that subset of taxa
- dist_permanova uses the distance matrix calculated by dist_calc, and doesn't use info from the otu_table

hope this helps?
David

**Comment by jorondo1** [comment]
Hi David, thanks for the quick reply. Yes, I meant the dist_calc ouptut. Since aitchison distances are computed on clr data I was acutally wondering why the dist_calc output contained sequence counts instead of clr-transformed table.

About filtering taxa; I'm asking about this because I neeed to remove unassigned genera after agglomerating to that taxrank, and in some samples this represents up to 15% of my estimated sequence counts. Given that CLR uses the mean of sequence counts to center the ratios, filtering before transformation would yield otu_table values that are different to the CLR-transformed original table (for the remaining taxa). 

I'm not sure how large that effect would be, but my understanding is that data should be CLR-transformed _before_ being subset, in order to preserve the ratio between taxa (subcompositional coherence, as far as I understand it). Intuitively, I would therefore first apply CLR, _then_ measure Aitchison distance on that filtered table. This is why I am confused by the dist_calc documentation, which requires the "identity"-tranformed table as input and therefore precludes filtering post CLR but pre dist_calc.

Let me know what you think ! cheers

p.s. unassigned genera that were assigned a species might sound weird; the output from Sourmash taxonomic assignment contains a lot of unknown taxa where the species is assigned as e.g. Streptococcaceae_bacterium, therefore containing _unassigned_ at the genus level.

**Reply by david-barnett** [reply]
> This is why I am confused by the dist_calc documentation, which requires the "identity"-tranformed table as input and therefore precludes filtering post CLR but pre dist_calc.

If I understand correctly, you could do the following steps: 
1. Compute CLR transformations using counts from all genera (taxonomically-classified or not) -> use tax_transform("clr")
2. Remove unclassified genera -> try e.g. tax_select(deselect = TRUE)
3. Compute the Euclidean distances between samples, using the remaining taxa -> dist_calc("euclidean")
4. Do permanova as discussed in your other thread

If you were not deselecting any taxa in between, then transforming to CLR and computing euclidean distances would be directly equivalent to aitchison distance on the original counts.

Doing the taxa deselection after CLR transformation means your distances won't strictly be Aitchison distances. 

I guess you're interested in assessing the influence of the taxa unclassified at genus level on your results, and this seems to me like a reasonable approach to assessing this

**Comment by jorondo1** [comment]
Ok that sounds great
I thought using the dist_calc("euclidean") was going to clr-transform a second time, since the doc requires no transformation beforehand

thanks for your help!

**Reply by david-barnett** [reply]
you're welcome, good luck :)

**Reply by jorondo1** [reply]
I spoke too fast, I should have simply tested it first. Applying dist_calc("aitchison") to clr--transformed table yields the following error:

```
> psGS_PERM <- psGenusSaliva %>% 
+   tax_transform(trans="rclr") %>% 
+   dist_calc(dist="robust.aitchison")
Error in `dist_calc()`:
! dist_calc 'aitchison' distance requires count data
ℹ your data are rclr-transformed (according to psExtra info)
ℹ see the ?dist_calc details section for more info
```

**Reply by david-barnett** [reply]
You want euclidean distance after the clr (or rclr) transformation

**Reply by jorondo1** [reply]
that's it, sorry for that! cheers
